src/manifolds/EssentialManifold.jl

```diff
@@ -0,0 +1,516 @@
+"""
+function dist_min_angle_pair_df_newton(m1, Φ1, c1, m2, Φ2, c2, t_min, t_low, t_high)
+    tol_dist = 1e-8
+    θ1 = 0
+    θ2 = 0
+    for i in 1:100
+        #compute auxiliary values
+        mc1 = m1 * cos(t_min + Φ1)
+        mc2 = m2 * cos(t_min + Φ2)
+
+        #compute θi, i=1,2
+        θ1 = acos(clamp(((m1 * sin(t_min + Φ1) + c1 - 1) / 2), -1, 1))
+        θ2 = acos(clamp(((m2 * sin(t_min + Φ2) + c2 - 1) / 2), -1, 1))
+
+        #compute the first derivatives di, i=1,2
+        s1 = 2 * sin(θ1)
+        s2 = 2 * sin(θ2)
+        d1 = (-θ1 * mc1) / s1
+        d2 = (-θ2 * mc2) / s2
+        d = d1 + d2
+
+        #compute the second derivatives ddi, i=1,2
+        eztuSq1 = (mc1 / s1)^2
+        eztuSq2 = (mc2 / s2)^2
+        dd1 = eztuSq1 + θ1 / 2 * cot(θ1 / 2) * (1 - eztuSq1)
+        dd2 = eztuSq2 + θ2 / 2 * cot(θ2 / 2) * (1 - eztuSq2)
+        dd = dd1 + dd2
+
+        #compute the new t_min
+        t_old = t_min
+        t_min = clamp(t_old - d / dd, t_low + tol_dist, t_high - tol_dist)
+        if abs(t_min - t_old) < tol_dist
+            break
+        end
+    end
+    f_min = θ1^2 + θ2^2
+    return t_min, f_min
+end
+
+@doc raw"""
+    manifold_dimension(M::EssentialManifold{is_signed, ℝ})
+
+Return the manifold dimension of the [`EssentialManifold`](@ref), which is `5`[^TronDaniilidis2017].
+"""
+function manifold_dimension(::EssentialManifold)
+    return 5
+end
+
+@doc raw"""
+    project(M::EssentialManifold, p, X)
+
+Project the matrix `X` onto the tangent space
+````math
+T_{p} \text{SO}(3)^2 = T_{\text{vp}}\text{SO}(3)^2 ⊕ T_{\text{hp}}\text{SO}(3)^2,
+````
+by first computing its projection onto the vertical space $T_{\text{vp}}\text{SO}(3)^2$ using [`vert_proj`](@ref).
+Then the orthogonal projection of `X` onto the horizontal space $T_{\text{hp}}\text{SO}(3)^2$ is defined as
+````math
+\Pi_h(X) = X - \frac{\text{vert\_proj}_p(X)}{2} \begin{bmatrix} R_1^T e_z \\ R_2^T e_z \end{bmatrix},
+````
+with $R_i = R_0 R'_i, i=1,2,$ where $R'_i$ is part of the pose of camera $i$ $g_i = (R'_i,T'_i) ∈ \text{SE}(3)$
+and $R_0 ∈ \text{SO}(3)$ such that $R_0(T'_2-T'_1) = e_z$.
+"""
+project(::EssentialManifold, ::Any, ::Any)
+
+function project!(M::EssentialManifold, Y, p, X)
+    X_proj = [pe' * Xe for (pe, Xe) in zip(p, X)]
+    X_proj .= [0.5 * (Xe - Xe') for Xe in X_proj]
+    # projection on the vertical component
+    s = vert_proj(M, p, X_proj)
+    # orthogonal projection
+    copyto!(
+        Y,
+        [
+            Xe - (s / 2) * get_vector(M.manifold, pe, pe[3, :], DefaultOrthogonalBasis())
+            for (Xe, pe) in zip(X_proj, p)
+        ],
+    )
+    return Y
+end
+
+"""
+    representation_size(M::EssentialManifold)
+
+Return the array dimensions required to represent an element on the
+[`EssentialManifold`](@ref) `M`, i.e. the vector of all array dimensions.
+"""
+@generated representation_size(::EssentialManifold) = (3, 3, 2)
```

Comment:
That would be correct for the `ArrayPowerRepresentation` but `NestedPowerRepresentation` doesn't have `representation_size`.

Reply:
Ah, thanks for noticing, I thought this was necessary for some allocations. Then we should remove this.